### PR TITLE
Allow overriding operation_id for each path on a single view function

### PR DESCRIPTION
**Pull request by gwaramadze**
We encountered this limitation on Flask. I hope you find it useful.

### CODE REVIEW

**Comment by yedpodtrzitko** [comment]
hi, can you please show the code/scenario where this happens?  thanks.

**Comment by gwaramadze** [comment]
@yedpodtrzitko sure! In a nutshell here is our scenario. We need to serve the same thing under two endpoints (for some time):

```python
@app.route("/api/v1/foo")  # this is a deprecated path
@app.route("/api/bar/v1/foo")  # we are moving APIs to this path
@api.validate(
    operation_id={
        "/api/v1/foo": "getFooDeprecated",
        "/api/bar/v1/foo": "getFoo",
    }
)
def foo(
```

**Comment by kemingy** [comment]
I'm thinking maybe we should have a `get_func_operation_id` method. So users can override this one. That might be more flexible. What do you think?

File: spectree/spec.py [hunk 1]
```diff
@@ -336,6 +335,19 @@ def _generate_spec(self) -> Dict[str, Any]:
         spec["security"] = get_security(self.config.security)
         return spec
 
+    def _get_func_operation_id(
```
[the remark this answers is not in the document]
Comment (gwaramadze) [reply spectree/spec.py:338]: Okay, got it. I will need more time for this as I had to switch contexts.

[the remark this answers is not in the document]
Comment (kemingy) [reply spectree/spec.py:338]: No worry. Take your time.

**Comment by kemingy** [comment]
Closing this since it's complete in #313

**Pull request closed without merging**